crates/fuel-core/src/executor.rs

```diff
@@ -528,93 +547,60 @@ where
         debug_assert!(block.transactions.is_empty());
         let mut iter = source.next(remaining_gas_limit).into_iter().peekable();
 
-        let mut coinbase_tx: Mint = match execution_kind {
-            ExecutionKind::DryRun => Default::default(),
-            ExecutionKind::Production => {
-                // The coinbase transaction should be the first.
-                // We will add actual amount of `Output::Coin` at the end of transactions execution.
-                Transaction::mint(
-                    TxPointer::new(block_height, Default::default()),
-                    vec![Output::coin(
-                        self.config.coinbase_recipient,
-                        0, // We will set it later
-                        *self.config.consensus_parameters.base_asset_id(),
-                    )],
-                )
-            }
-            ExecutionKind::Validation => {
-                let mint = if let Some(MaybeCheckedTransaction::Transaction(
-                    Transaction::Mint(mint),
-                )) = iter.next()
-                {
-                    mint
-                } else {
-                    return Err(ExecutorError::CoinbaseIsNotFirstTransaction)
-                };
-                self.check_coinbase(block_height, mint, None)?
-            }
-        };
-
-        let mut tx_index = if execution_kind != ExecutionKind::DryRun {
-            // Skip the coinbase transaction.
-            block.transactions.push(coinbase_tx.clone().into());
-            1
-        } else {
-            0
-        };
+        let mut execute_transaction = |execution_data: &mut ExecutionData,
```

Comment:
It seems like it'd be cleaner to move the logic in this closure to the execute_transaction fn. Is there any reason why we need a closure for execute_transaction that wraps calls to the actual `fn execute_transction()`?

Reply:
There are two main reason: 

1) We want to get a result from transaction execution and based on the result, decide what we need to do with it(skip or error, commit or not):
<img width="886" alt="image" src="https://github.com/FuelLabs/fuel-core/assets/18346821/c121e716-f774-4684-a3c1-ff5aae2a060b">

2) It pushes the transaction to the block and increases the tx counter:
<img width="596" alt="image" src="https://github.com/FuelLabs/fuel-core/assets/18346821/89844739-9f4b-4eb0-bec2-be520bcaeb1a">